crates/worker/src/purgatory.rs

```diff
@@ -171,3 +195,80 @@ impl Inner {
         });
     }
 }
+
+struct GlobalState {
+    db_client: DynamoDBClient,
+    s3_client: S3Client,
+    pub items: Vec<Item>,
+}
+
+impl GlobalState {
+    pub fn new(db_client: DynamoDBClient, s3_client: S3Client) -> Self {
+        Self {
+            db_client,
+            s3_client,
+            items: vec![],
+        }
+    }
+
+    pub async fn sync(&mut self, sync_from: &DateTime<Utc>) -> Result<(), GlobalPurgeError> {
+        const MAX_CAPACITY: usize = 1000;
+
+        let mut last_evaluated_key = None;
```

Comment:
How do we use it except updating?

Reply:
In case the result of query exceeds SOME_LIMIT DynamoDb will stop and return current results + last_evaluated_key so you can continue your query starting from that point